bundles/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/internal/HarmonyHubDynamicTypeProvider.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.harmonyhub.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.storage.StorageService;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.AbstractStorageBasedTypeProvider;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link HarmonyHubDynamicTypeProvider} is an instance of a {@link AbstractStorageBasedTypeProvider} for the
+ * HarmonyHub
+ * binding
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@Component(service = { HarmonyHubDynamicTypeProvider.class, ChannelTypeProvider.class })
+@NonNullByDefault
+public class HarmonyHubDynamicTypeProvider extends AbstractStorageBasedTypeProvider {
+
+    @Activate
+    public HarmonyHubDynamicTypeProvider(@Reference StorageService storageService) {
+        super(storageService);
+    }
+
+    public void removeChannelTypesForThing(ThingUID uid) {
```

Comment:
Isn't it possible and useful to put that method in the abstract class (core framework)) ?

Reply:
No, because the core class also works for thing-types, where this method should to make sense.

Reply:
What is the problem if the method is properly documented ?

If not, I understand that every binding implementing a channel type provider will have to duplicate this method ?

Reply:
Side comment: Can it be specialized into a subclass in core?

Reply:
@lolodomo IMO it's bad design to implement methods in a generic super-class that only apply to some specialized cases. It doesn't even apply to all `ChannelTypeProviders` because it is not necessary to have them "per-thing", you could also make them available for all things, especially if you don't have a thing-type/thing but a limited set of dynamic thing-types with dynamic channel-types.

@jlaur That would be possible. But on the other hand: It's a pretty straight-forward two-liner. I don't think its necessary to create a class for that.

Reply:
By the way, I realized that this method is in fact very specific to this binding and will not be required by all bindings.